### PR TITLE
Change conda related sections in the README and the docs

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Small improvements of the conda installation instructions.

### CODE REVIEW

File: doc/quickstart.rst [hunk 1]
```diff
@@ -8,10 +8,10 @@ CadQuery QuickStart
 
 Want a quick glimpse of what CadQuery can do?  This quickstart will demonstrate the basics of CadQuery using a simple example
 
-Prerequisites: Anaconda/Miniconda + CQ-editor running from an environment
+Prerequisites: conda + CQ-editor running from an environment
```
Comment (Jojain) [inline doc/quickstart.rst:11]: Sharing what goes through my head, but couldn't it be possible to setup a [binder](https://mybinder.org/) environment to give quick example of CQ to people ?

Comment (adam-urbanczyk) [reply doc/quickstart.rst:11]: I think that we tried, but it was really slow

Comment (Jojain) [reply doc/quickstart.rst:11]: Ah ok didn't know, nevermind then

File: doc/installation.rst [hunk 2]
```diff
@@ -5,11 +5,29 @@ Installing CadQuery
 
 CadQuery 2 is based on
 `OCP <https://github.com/CadQuery/OCP>`_,
-which is a set of Python bindings for the open-source `OpenCascade <http://www.opencascade.com/>`_ modelling kernel.
+which is a set of Python bindings for the open-source ` OpenCascade <http://www.opencascade.com/>`_ modelling kernel.
```
Comment (jmwright) [inline doc/installation.rst:8]: Why the space in front of "OpenCascade" here?

**Comment by jmwright** [comment]
@adam-urbanczyk This PR looks good to me except that I haven't used miniforge, so I don't know it's strengths/weaknesses vs miniconda.

File: doc/installation.rst (hunk 2 above)
Comment (adam-urbanczyk) [reply doc/installation.rst]: Typo, fixed

**Comment by adam-urbanczyk** [comment]
@jmwright miniforge is a minimal conda installation configured to use the (free) conda-forge channel (which is the channel we are relying on). I also added this based on the comments on bloat floating around HaD/HN. I also added section on portable win install.

**Comment by jmwright** [comment]
Ok, thanks @adam-urbanczyk  .

@Jojain since you're already involved in this discussion, will you be the other official reviewer? We normally try to have two devs check a PR before merging unless it's a super simple one.

**Comment by Jojain** [comment]
Sure if you want too, I will try this new installation process

**Comment by jmwright** [comment]
@Jojain If you have time, please do.

**Comment by Jojain** [comment]
@adam-urbanczyk 
I just tried the miniforge commands lines without success.
`curl.exe -o miniforge.exe https://superuser.com` seems to install something but without more info (I don't know what is curl)

Then
`start /wait "" miniforge.exe /InstallationType=JustMe /RegisterPython=0 /NoRegistry=1 /NoScripts=1 /S /D=%USERPROFILE%\Miniforge3
` Fails with : System cannot execute specified program

**Comment by jmwright** [comment]
It installs fine for me in Ubuntu, so maybe the problem is specific to Windows?

File: README.md [hunk 3]
```diff
@@ -58,17 +58,27 @@ conda install -c conda-forge -c cadquery cadquery=master
...
 ```
-# Install the script to ~/miniconda
-wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
-bash miniconda.sh -b -p $HOME/miniconda
+# Install the script to ~/miniforge
+wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh -O miniforge.sh
+bash miniforge.sh -b -p $HOME/miniforge
 
 # To activate and use Miniconda
 source $HOME/miniconda/bin/activate
```
Comment (jmwright) [inline README.md:70]: I think that `miniconda` in this line has to change to `miniforge`.

File: doc/installation.rst [hunk 4]
```diff
@@ -7,9 +7,27 @@ CadQuery 2 is based on
...
 ----------------------------------------------
-CadQuery requires OCP and Python version 3.x
+In principle any conda distrubtion will work, but it is probably best to install `Miniforge <https://github.com/conda-forge/miniforge>`_ to a local directory and to avoid running `conda init`. After performing a local directory installation, Miniforge can be activated via the [scripts,bin]/activate scripts. This will help avoid polluting and breaking the local Python installation. In Linux/MacOS, the local directory installation method looks something like this::
+
+        # Install the script to ~/miniforge
+        wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh -O miniforge.sh
+        bash miniforge.sh -b -p $HOME/miniforge
+
+        # To activate and use Miniconda
+        source $HOME/miniconda/bin/activate
```
Comment (jmwright) [inline doc/installation.rst:19]: Same comment as above. I think that `miniconda` in this line has to change to `miniforge`.

**Comment by jmwright** [comment]
The installation of miniforge works on Ubuntu, but the environment solve always fails when I try to run the following line.
```
conda install -c conda-forge -c cadquery cadquery=master
```

**Comment by adam-urbanczyk** [comment]
> @adam-urbanczyk I just tried the miniforge commands lines without success. `curl.exe -o miniforge.exe https://superuser.com` seems to install something but without more info (I don't know what is curl)
> 
> Then `start /wait "" miniforge.exe /InstallationType=JustMe /RegisterPython=0 /NoRegistry=1 /NoScripts=1 /S /D=%USERPROFILE%\Miniforge3 ` Fails with : System cannot execute specified program

I made a stupid copy-paster error - should be corrected now @Jojain .

File: doc/installation.rst [hunk 5]
```diff
@@ -7,9 +7,27 @@ CadQuery 2 is based on
...
+On Windows one can install locally as follows::
+
+        :: Install
+        curl -L -o miniforge.exe https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Windows-x86_64.exe
+        start /wait "" miniforge.exe /InstallationType=JustMe /RegisterPython=0 /NoRegistry=1 /NoScripts=1 /S /D=%USERPROFILE%\Miniforge3
+
+        :: Activate
+        cmd /K ""%USERPROFILE%/Miniforge3/Scripts/activate.bat" "%USERPROFILE%/Miniforge3""
+
+It might be worthwhile to consider using ``/NoScipts=0`` to add an activation shortcut.
```
Comment (Jojain) [inline doc/installation.rst:30]: typo ``/NoScipts=0`` -> ``/NoScripts=0``

It's however not clear how this relate to a shortcut and how to setup one ?

**Comment by Jojain** [comment]
I got the install successful. I've read very quickly what is miniforge, if I understand well (put aside what's under the hood) miniforge is miniconda with the default channel setup to `conda-forge` ? So one could install cadquery with the following :
`conda install -c cadquery cadquery=master` ?

Since I have also miniconda installed I don't know which one is activating but if I run 
`conda activate base` in a cmd prompt does it activate the miniforge env or I need to use your command line (for activation?)

**Comment by jmwright** [comment]
@Jojain On Linux, I run something like `$HOME/miniforge/bin/activate` to make sure I have the right one (notice the path has `miniforge` and not `minconda`. That's assuming that miniconda is not already activated for the current shell.

**Comment by jmwright** [comment]
I swapped Anaconda out for Miniforge on my main Ubuntu 20.04 install and it works fine. However, the solve still always fails on the Lubuntu 20.04 VM that I've also tried it on.  Below is what it always shows.

Screenshot from 2022-02-08 08-13-43

**Comment by adam-urbanczyk** [comment]
> 

Can you share (link to)  the vm image?

**Comment by jmwright** [comment]
VM image download is [here](https://cdimage.ubuntu.com/lubuntu/releases/20.04.3/release/lubuntu-20.04.3-desktop-amd64.iso). I ran it in VirtualBox.

**Comment by lorenzncode** [comment]
@jmwright This could be an out of memory issue.  I've also experienced that with `conda` when allocating too little memory.  I've had it fail when allocating 4GB.  I couldn't reproduce that today at 4GB.  Here at 2GB (`dmesg`):

Screenshot from 2022-02-08 23-11-49

One advantage of miniforge is it also provides mamba.  You might try replacing `conda install` with `mamba install` and see if it completes successfully.  

I checked RSS of the conda/mamba install process today on Ubuntu 20.04.
conda:   peak 2.7GiB  (separate install from above, with more memory allocated)
mamba: peak 280MiB

**Comment by Jojain** [comment]
I don't really want to break my miniconda install to try miniforge alone, but I have been successful with the installation and activate on my Windows computer, so I think it's good (I've also read it's bundled directly with mamba ?)

**Comment by jmwright** [comment]
If `mamba` uses less memory, should we consider using it by default instead of the `conda` command? I'm not familiar with `mamba`, so I don't know the implications.

**Comment by Jojain** [comment]
I don't know about the memory but it's faster than conda and gives better error message when there is dependencies conflicts (however for some reason ``mamba activate myenv` doesn't work so I use both conda and mamba )

**Comment by fedorkotov** [comment]
I have just nuked minicaonda and replaced it with mambaforge (condaforge variant with mamba preinstalled in base environment). It is fast. And `mamba activate envname` works. But importing cadquery fails see #990. Don't know if mamba or condaforge have anything to do with this problem though. My environment for cadquery in miniconda (that is now destroyed) worked but it was not updated for a long time.

File: doc/installation.rst (hunk 4 above)
Comment (adam-urbanczyk) [reply doc/installation.rst]: done

File: doc/installation.rst (hunk 5 above)
Comment (adam-urbanczyk) [reply doc/installation.rst]: Done, this line adds a shortcut to the start menu

**Comment by adam-urbanczyk** [comment]
@jmwright @Jojain I think I fixed all the defects. Do you want to change from miniforge to mambaforge or leave as-is?

BTW: I cannot reproduce #990.

**Comment by jmwright** [comment]
I just switched to mambaforge and it worked fine to create a CadQuery environment, and `import cadquery` worked as well. I'm on Linux though (Ubuntu 20.04).

**Comment by jmwright** [comment]
I am also unable to reproduce #990 on Windows 11 64-bit with mambaforge. I'm ok with using mambaforge, but again, don't have a lot of experience with it. I have switch my systems over to it and will be using it as my primary way to handle my CadQuery environments moving forward.

**Comment by adam-urbanczyk** [comment]
> I am also unable to reproduce #990 on Windows 11 64-bit with mambaforge. I'm ok with using mambaforge, but again, don't have a lot of experience with it. I have switch my systems over to it and will be using it as my primary way to handle my CadQuery environments moving forward.

OK, so I'd propose to stay with conda for now. The only known issue is the memory usage, I think that it is acceptable.

**Comment by fedorkotov** [comment]
For me a workaround for #990 on Windows 8.1 was adding `-c conda-forge` to installation or environment creation command. No one seems to be able to reproduce it on more modern operating systems (Windows or Linux). Besides Windows 8.1 has only one more year of life left before end of extended support from M$. So this is not really an issue anymore.

**Comment by adam-urbanczyk** [comment]
Alright merging with conda/miniforge, in the future we'll probably change it to mamba/mambaforge.

**Pull request merged**